### ISSUE TITLE
torch1.10 IndexError: tuple index out of range

### ISSUE THREAD

**Issue by shizhediao** [NVIDIA/apex#1303]
Similar to https://github.com/NVIDIA/apex/pull/1282 I met an issue:
```    
if cached_x.grad_fn.next_functions[1][0].variable is not x:
IndexError: tuple index out of range
 ```

I tried the fix but the time cost is 10 times.

It works well for torch1.9 but this issue arises when use torch 1.10

**Comment by wptoux** [comment]
For a temp fix, I just comment out the whole if check and everything goes well ~

**Comment by shizhediao** [comment]
Thanks. I solved it according to #1282.
